src/lib/y2storage/boot_requirements_checker.rb

```diff
@@ -61,26 +67,36 @@ def needed_partitions(target = :desired)
 
   protected
 
-    attr_reader :settings
+    # @return [Devicegraph] starting situation.
     attr_reader :devicegraph
 
+    # @return [Array<Planned::Device>] devices that are already planned to be
+    #   added to the starting devicegraph.
+    attr_reader :planned_devices
+
+    # @return [String, nil] device name of the disk that the system will try to
+    #   boot first. Only useful in some scenarios like legacy boot.
```

Comment:
please document what happens if there is nil. How it then pick booting disk? auto-detection or simple picking first one?